bin/commander.js

```diff
@@ -17,255 +17,133 @@
- *
- * @return {string[]} List of files.
+ * Internal dependencies
  */
-function getIgnoredFiles() {
-  const maybeIgnoredFiles = readFileSync(DISTIGNORE_FILE, 'utf8').split('\n');
-  const ignoredFiles = [];
-
-  for (const file of maybeIgnoredFiles) {
-    if (file.startsWith('#')) {
-      continue;
-    }
-
-    if (file.trim() === '') {
-      continue;
-    }
-
-    const fileFromRoot = PLUGIN_DIR + '/' + file;
-
-    if (!existsSync(fileFromRoot)) {
-      continue;
-    }
-
-    if (lstatSync(fileFromRoot).isDirectory()) {
-      ignoredFiles.push(`${file}/`);
-    } else {
-      ignoredFiles.push(file);
-    }
-  }
-
-  return ignoredFiles;
-}
-
-function createBuildDir() {
-  const buildPath = 'build/web-stories';
-
-  // Copy plugin folder to temporary location.
-  execSync(`mkdir -p ${buildPath}`, {
-    stdio: ['pipe', 'pipe', 'ignore'],
-  });
-}
-
-function copyFiles(skipVendor) {
-  const ignoredFiles = getIgnoredFiles();
-
-  if (skipVendor) {
-    ignoredFiles.push('vendor/');
-  }
-
-  const excludeList = ignoredFiles
-    .map((file) => `--exclude '${file}'`)
-    .join(' ');
-
-  const buildPath = 'build/web-stories';
+import {
+  bundlePlugin,
+  buildFonts,
+  getCurrentVersionNumber,
+  updateVersionNumbers,
+  updateAssetsURL,
+} from './utils/index.js';
 
-  // Copy plugin folder to temporary location.
-  execSync(`rsync -a ${excludeList} ${PLUGIN_DIR}/ ${buildPath}/`, {
-    stdio: ['pipe', 'pipe', 'ignore'],
-  });
-}
-
-function deleteExistingZipFiles() {
-  execSync(`rm -rf web-stories-*.zip`, {
-    cwd: PLUGIN_DIR + '/build',
-    stdio: ['pipe', 'pipe', 'ignore'],
-  });
-}
-
-function deleteExistingBuildFolder() {
-  execSync(`rm -rf web-stories`, {
-    cwd: PLUGIN_DIR + '/build',
-    stdio: ['pipe', 'pipe', 'ignore'],
-  });
-}
-
-function generateZipFile(filename) {
-  const pluginFileContent = readFileSync(PLUGIN_FILE, 'utf8');
-  const currentVersion = pluginFileContent
-    .match(VERSION_CONSTANT_REGEX)[1]
-    .trim();
-
-  const zipName = filename || `web-stories-${currentVersion}.zip`;
+const PLUGIN_DIR = process.cwd();
+const PLUGIN_FILE = `${PLUGIN_DIR}/web-stories.php`;
+const README_FILE = `${PLUGIN_DIR}/readme.txt`;
+const PLUGIN_BUILD_DIR = `${PLUGIN_DIR}/build/web-stories`;
+const FONTS_FILE = PLUGIN_DIR + '/includes/data/fonts.json';
 
-  // This ensures the folder in the final ZIP file is always named "web-stories".
-  execSync(`zip -r ${zipName} web-stories`, {
-    cwd: PLUGIN_DIR + '/build',
-    stdio: ['pipe', 'pipe', 'ignore'],
+program
+  .command('version')
+  .arguments('[version]')
+  .option(
+    '--nightly',
+    'Whether this is a nightly build and thus should append the current revision to the version number.'
+  )
+  .description('Bump the version of the plugin')
+  .on('--help', () => {
+    console.log('');
+    console.log('Examples:');
+    console.log('  # Prepare stable release');
+    console.log('  $ commander.js version 1.2.3');
```

Comment:
Can we define the version number in a json file or something similar and not hardcode it like this.

Reply:
The main plugin file is currently the source of truth for the version number. If you wanna bump the version, use this command. How is that hardcoded?

Reply:
For clarification, the `commander.js version 1.2.3` example here means:

_I wanna change this plugin's version from whatever it currently is to `1.2.3`_